lib/webmention/client.rb

```diff
@@ -93,84 +85,87 @@ def self.send_mention endpoint, source, target, full_response=false
+          endpoint = discover_webmention_endpoint_from_header(link_headers)
+
           return endpoint if endpoint
         end
 
         # Do we support webmention?
-        endpoint = self.discover_webmention_endpoint_from_html response.body.to_s
+        endpoint = discover_webmention_endpoint_from_html(response.body.to_s)
+
         return endpoint if endpoint
 
         # TODO: Move to supports_pingback? method
         # Last chance, do we support Pingback?
         # if !doc.css('link[rel="pingback"]').empty?
         #   return doc.css('link[rel="pingback"]').attribute("href").value
         # end
-
       rescue EOFError
       rescue Errno::ECONNRESET
       end
 
-      return false
+      false
     end
 
-    def self.discover_webmention_endpoint_from_html html
+    def self.discover_webmention_endpoint_from_html(html)
       doc = Nokogiri::HTML(html)
+
       if !doc.css('[rel~="webmention"]').css('[href]').empty?
-        doc.css('[rel~="webmention"]').css('[href]').attribute("href").value
+        doc.css('[rel~="webmention"]').css('[href]').attribute('href').value
       elsif !doc.css('[rel="http://webmention.org/"]').css('[href]').empty?
-        doc.css('[rel="http://webmention.org/"]').css('[href]').attribute("href").value
+        doc.css('[rel="http://webmention.org/"]').css('[href]').attribute('href').value
       elsif !doc.css('[rel="http://webmention.org"]').css('[href]').empty?
-        doc.css('[rel="http://webmention.org"]').css('[href]').attribute("href").value
+        doc.css('[rel="http://webmention.org"]').css('[href]').attribute('href').value
       else
         false
       end
     end
 
-    def self.discover_webmention_endpoint_from_header header
-      if matches = header.match(%r{<([^>]+)>; rel="[^"]*\s?webmention\s?[^"]*"})
+    def self.discover_webmention_endpoint_from_header(header)
+      if (matches = header.match(/<([^>]+)>; rel="[^"]*\s?webmention\s?[^"]*"/))
+        return matches[1]
+      elsif (matches = header.match(/<([^>]+)>; rel=webmention/))
         return matches[1]
-      elsif matches = header.match(%r{<([^>]+)>; rel=webmention})
-          return matches[1]
-      elsif matches = header.match(%r{rel="[^"]*\s?webmention\s?[^"]*"; <([^>]+)>})
+      elsif (matches = header.match(/rel="[^"]*\s?webmention\s?[^"]*"; <([^>]+)>/))
         return matches[1]
-      elsif matches = header.match(%r{rel=webmention; <([^>]+)>})
+      elsif (matches = header.match(/rel=webmention; <([^>]+)>/))
         return matches[1]
-      elsif matches = header.match(%r{<([^>]+)>; rel="http://webmention\.org/?"})
+      elsif (matches = header.match(%r{<([^>]+)>; rel="http://webmention\.org/?"}))
         return matches[1]
-      elsif matches = header.match(%r{rel="http://webmention\.org/?"; <([^>]+)>})
+      elsif (matches = header.match(%r{rel="http://webmention\.org/?"; <([^>]+)>}))
         return matches[1]
       end
-      return false
+
+      false
     end
```

Comment:
I recently created [the webmention-endpoint-ruby gem](https://github.com/jgarber623/webmention-endpoint-ruby) to handle endpoint discovery. In a future PR, I'd like to abstract away this functionality by using that gem. For another day, though…